### PR TITLE
Add param to skip external ACL updates

### PR DESCRIPTION
**Pull request by MrCreosote**
External ACL updates can take a significant portion of the data fetch time, especially when `no_data=1`.

Also useful for admins that don't want to add themselves to the ACL lists.

Fixes https://github.com/kbase/workspace_deluxe/issues/486

### CODE REVIEW

**Comment by MrCreosote** [comment]
Closed / reopened to get Travis to run

File: src/us/kbase/workspace/test/kbase/JSONRPCLayerTest.java [hunk 1]
```diff
@@ -904,7 +904,9 @@ public void saveAndGetObjects() throws Exception {
 				.withName("auto1");
 		Map<String, List<String>> exp = new HashMap<String, List<String>>();
 		ObjectData objo = CLIENT1.getObjects2(new GetObjects2Params()
-			.withObjects(toObjSpec(Arrays.asList(ojbid)))).getData().get(0);
+			.withObjects(toObjSpec(Arrays.asList(ojbid)))
+			.withSkipExternalAclUpdates(1L) // should have no effect
+			).getData().get(0);
```
Comment (ialarmedalien) [inline src/us/kbase/workspace/test/kbase/JSONRPCLayerTest.java:909]: is there another test where you verify that adding the skip external ACL updates param has no effect, either implicitly or explicitly?

File: src/us/kbase/workspace/kbase/ArgUtils.java [hunk 2]
```diff
@@ -395,21 +393,34 @@ private static String formatDate(final Instant date) {
...
 			}
 			final PermError error = makeExternalIDsReadable(o, permHandler);
-			final ByteArrayFileCache resource = o.getSerializedData();
+			final UObject data;
+			try {
+				data = o.getSerializedData() == null ? null : o.getSerializedData().getUObject();
+			} catch (IOException e) {
+				// impossible to test in integration tests, shouldn't occur
+				throw new RuntimeException(
+						"An unexpected error occured: " + e.getLocalizedMessage(), e);
```
Comment (ialarmedalien) [inline src/us/kbase/workspace/kbase/ArgUtils.java:420]: s/occured/occurred/

File: src/us/kbase/workspace/test/kbase/JSONRPCLayerTest.java (hunk 1 above)
Comment (MrCreosote) [reply src/us/kbase/workspace/test/kbase/JSONRPCLayerTest.java:909]: there's not really another way to test this, I think. The current tests are:

* when no external systems are connected to the workspace, the new param does nothing. In that case, there's nothing to query to ensure nothing happened (this is the test above).
* when external systems are connected (in the handle/shock test and the samples test), getting objects is tested with and without the new parameter to ensure it works as expected and the ACLs are updated or not.

I'm not seeing another way to test off the top of my head, anyway

File: src/us/kbase/workspace/kbase/ArgUtils.java (hunk 2 above)
Comment (MrCreosote) [reply src/us/kbase/workspace/kbase/ArgUtils.java]: fixed

**Review by ialarmedalien — COMMENTED** [review]
Merge away!

**Pull request merged**